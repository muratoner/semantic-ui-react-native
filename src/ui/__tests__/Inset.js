import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import {StyleSheet} from 'react-native'
import {Inset} from '../../'

describe('Inset Component', () => {
	it('should render without issues', () => {
		const component = shallow(<Inset all={10} style={{backgroundColor: 'red'}} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('padding', 10)
		expect(style).toHaveProperty('backgroundColor', 'red')
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with paddingTop size 10', () => {
		const component = shallow(<Inset top={10} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('paddingTop', 10)
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with paddingBottom size 10', () => {
		const component = shallow(<Inset bottom={10} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('paddingBottom', 10)
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with paddingLeft size 10', () => {
		const component = shallow(<Inset left={10} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('paddingLeft', 10)
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with paddingRight size 10', () => {
		const component = shallow(<Inset right={10} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('paddingRight', 10)
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with paddingHorizontal size 10', () => {
		const component = shallow(<Inset horizontal={10} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('paddingHorizontal', 10)
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with paddingVertical size 10', () => {
		const component = shallow(<Inset vertical={10} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('paddingVertical', 10)
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with flex', () => {
		const component = shallow(<Inset flex />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('flex', true)
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with flex size is 2', () => {
		const component = shallow(<Inset flex={2} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('flex', 2)
		expect(toJson(component)).toMatchSnapshot()
	})
})
