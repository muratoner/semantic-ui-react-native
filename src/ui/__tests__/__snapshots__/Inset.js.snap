// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inset Component should render with flex 1`] = `
<View
  style={
    Array [
      Object {
        "flex": true,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      undefined,
    ]
  }
/>
`;

exports[`Inset Component should render with flex size is 2 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 2,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      undefined,
    ]
  }
/>
`;

exports[`Inset Component should render with paddingBottom size 10 1`] = `
<View
  style={
    Array [
      Object {
        "flex": undefined,
        "padding": undefined,
        "paddingBottom": 10,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      undefined,
    ]
  }
/>
`;

exports[`Inset Component should render with paddingHorizontal size 10 1`] = `
<View
  style={
    Array [
      Object {
        "flex": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": 10,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      undefined,
    ]
  }
/>
`;

exports[`Inset Component should render with paddingLeft size 10 1`] = `
<View
  style={
    Array [
      Object {
        "flex": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": 10,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      undefined,
    ]
  }
/>
`;

exports[`Inset Component should render with paddingRight size 10 1`] = `
<View
  style={
    Array [
      Object {
        "flex": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": 10,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      undefined,
    ]
  }
/>
`;

exports[`Inset Component should render with paddingTop size 10 1`] = `
<View
  style={
    Array [
      Object {
        "flex": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": 10,
        "paddingVertical": undefined,
      },
      undefined,
    ]
  }
/>
`;

exports[`Inset Component should render with paddingVertical size 10 1`] = `
<View
  style={
    Array [
      Object {
        "flex": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`Inset Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "flex": undefined,
        "padding": 10,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
